### ISSUE TITLE
Pub/Sub gRPC CA crt file is no longer valid

### ISSUE THREAD

**Issue by aperkins1310** [akka/alpakka#1878]
### Versions used 
"com.lightbend.akka" % "akka-stream-alpakka-google-cloud-pub-sub-grpc" % "1.1.0"
Akka version: 
2.5.23

### Expected Behavior
The Pub/Sub connector should be able to connect to pub sub via https and complete the SSL handshake

### Actual Behavior
Google is currently rolling out a set of new certs. When handshaking with one of the new certs,
a `io.grpc.StatusRuntimeException: UNAVAILABLE: io exception` is thrown. This currently doesn't happen all the time, and is dependent on the certificate returned by the particular pub sub instance. This seems to be somewhat location based, as locally it flips between the old and new cert, and in GCP zones `europe-west1-b`, `europe-west1-c`and `europe-west1-d` have been more severely impacted during different times during the rollout.
We have confirmed with google through a quite lengthy support process over the last couple of days that this is the case, and forcing the use of a valid CA .crt file, such as that found in our docker base image at `/etc/ssl/certs/ca-certificates.crt` has resolved the issue.

### Relevant logs
The error first gets hidden behind a misleading cause:
`Caused by: io.grpc.netty.shaded.io.netty.channel.AbstractChannel$AnnotatedSocketException: Network is unreachable: pubsub.googleapis.com/2a00:1450:400c:c08:0:0:0:5f:443` in GCP, or locally a 'no route to host'
Forcing the use of  ipv4 with the JAVA_OPT `-Djava.net.preferIPv4Stack=true` then shows the underlying error cause:
`javax.net.ssl.SSLHandshakeException: No trust manager was able to validate this certificate chain`
This is because the default `GoogleInternetAuthorityG3.crt` bundled in the module is no longer valid with the new certs.

### Reproducible Test Case
Connecting to a pub sub with the new cert:
```
echo | openssl s_client -showcerts -servername pubsub.googleapis.com -connect pubsub.googleapis.com:443 2>/dev/null | openssl x509 -inform pem -noout -text
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            96:2f:4c:f4:63:23:c4:9f:08:00:00:00:00:0d:c9:b3
    Signature Algorithm: sha256WithRSAEncryption
        Issuer: C=US, O=Google Trust Services, CN=GTS CA 1O1
        Validity
            Not Before: Jul 29 17:27:05 2019 GMT
            Not After : Oct 27 17:27:05 2019 GMT
        Subject: C=US, ST=California, L=Mountain View, O=Google LLC, CN=edgecert.googleapis.com
        Subject Public Key Info:
            Public Key Algorithm: id-ecPublicKey
                Public-Key: (256 bit)
                pub:
                    04:e2:17:a2:68:9c:16:0e:ab:e9:8a:03:31:a2:79:
                    61:3c:af:87:6d:0b:f4:dc:88:69:3a:f6:1e:24:80:
                    cc:d5:7c:c1:1d:ca:11:88:c7:ef:83:e5:4f:b9:8f:
                    93:fc:0b:7e:e5:03:1a:b7:40:79:04:88:6e:ba:0e:
                    6e:75:05:4b:14
                ASN1 OID: prime256v1
                NIST CURVE: P-256
        X509v3 extensions:
            X509v3 Key Usage: critical
                Digital Signature
            X509v3 Extended Key Usage:
                TLS Web Server Authentication
            X509v3 Basic Constraints: critical
                CA:FALSE
            X509v3 Subject Key Identifier:
                8F:A2:59:A4:11:E6:F2:6A:0D:8B:8C:9B:16:55:61:6D:01:04:FE:06
            X509v3 Authority Key Identifier:
                keyid:98:D1:F8:6E:10:EB:CF:9B:EC:60:9F:18:90:1B:A0:EB:7D:09:FD:2B

            Authority Information Access:
                OCSP - URI:http://ocsp.pki.goog/gts1o1
                CA Issuers - URI:http://pki.goog/gsr2/GTS1O1.crt

            X509v3 Subject Alternative Name:
                DNS:edgecert.googleapis.com, DNS:android.googleapis.com, DNS:bigtable.googleapis.com, DNS:clouderrorreporting.googleapis.com, DNS:cloudtrace.googleapis.com, DNS:container.googleapis.com, DNS:dataflow.googleapis.com, DNS:datastore.googleapis.com, DNS:fcm.googleapis.com, DNS:firestore.googleapis.com, DNS:logging.googleapis.com, DNS:monitoring.googleapis.com, DNS:pubsub.googleapis.com, DNS:runtimeconfig.googleapis.com, DNS:servicecontrol.googleapis.com, DNS:spanner.googleapis.com, DNS:stackdriver.googleapis.com
            X509v3 Certificate Policies:
                Policy: 2.23.140.1.2.2
                Policy: 1.3.6.1.4.1.11129.2.5.3

            X509v3 CRL Distribution Points:

                Full Name:
                  URI:http://crl.pki.goog/GTS1O1.crl

            CT Precertificate SCTs:
                Signed Certificate Timestamp:
                    Version   : v1(0)
                    Log ID    : 63:F2:DB:CD:E8:3B:CC:2C:CF:0B:72:84:27:57:6B:33:
                                A4:8D:61:77:8F:BD:75:A6:38:B1:C7:68:54:4B:D8:8D
                    Timestamp : Jul 29 18:27:06.092 2019 GMT
                    Extensions: none
                    Signature : ecdsa-with-SHA256
                                30:46:02:21:00:BF:97:DA:A3:4A:26:91:25:11:1C:45:
                                00:F0:CD:F3:8A:A6:F9:08:8F:F1:6C:82:A0:F1:E2:4A:
                                87:BF:7A:E9:A4:02:21:00:FD:73:A2:F5:2B:74:8F:F4:
                                91:90:7E:42:82:74:F3:31:EC:EE:12:71:7F:E4:C6:FD:
                                DA:43:09:9E:89:60:30:EA
                Signed Certificate Timestamp:
                    Version   : v1(0)
                    Log ID    : 74:7E:DA:83:31:AD:33:10:91:21:9C:CE:25:4F:42:70:
                                C2:BF:FD:5E:42:20:08:C6:37:35:79:E6:10:7B:CC:56
                    Timestamp : Jul 29 18:27:06.378 2019 GMT
                    Extensions: none
                    Signature : ecdsa-with-SHA256
                                30:44:02:20:2F:70:93:6C:A6:51:EA:14:8D:B9:E1:CB:
                                12:16:B3:2C:C4:6E:DC:29:B6:9C:A3:0D:C4:5F:EF:38:
                                6C:97:AB:B0:02:20:24:6E:B7:64:54:C1:BD:AF:AB:E4:
                                AC:33:19:28:49:4E:16:0F:72:27:C2:A5:A7:F3:BC:FF:
                                C0:24:ED:FE:5F:D5
    Signature Algorithm: sha256WithRSAEncryption
         a8:f3:ec:d7:86:67:34:1a:8d:87:de:d4:05:fa:b2:dd:9f:98:
         6f:a7:18:73:b7:fc:6e:4e:88:b4:6b:ba:ca:5f:6c:10:d0:8a:
         56:68:5b:79:a8:cb:3c:62:fa:62:ac:a9:3e:5c:d0:64:5a:8b:
         dc:37:2e:8d:2d:30:e6:0e:4c:97:8b:bc:29:f1:50:0c:b6:12:
         ce:75:2f:a6:48:e4:9b:38:b6:d8:6f:33:09:63:0c:9c:e3:72:
         15:0c:6d:0c:96:09:bd:27:21:b3:c7:1a:4a:82:30:3f:46:fc:
         8b:3a:ea:0f:90:66:e8:ad:bc:93:22:30:08:b1:d6:11:60:8c:
         28:6c:cb:f9:e8:ad:78:91:ca:7a:70:35:9e:1a:f3:5e:1b:7c:
         d5:2a:be:da:2f:e1:a1:7a:0a:9a:9a:70:43:e9:94:fe:7a:c8:
         19:ed:fa:6d:b3:e6:9f:d8:26:42:1f:ee:07:8e:58:98:bd:f4:
         ef:df:1b:02:0d:8d:23:ef:a8:5d:d1:e5:67:0f:b5:6d:2c:8f:
         28:69:d5:c6:dc:aa:b4:f3:78:5c:e5:5d:9a:cd:0b:76:60:42:
         dc:41:4b:ab:62:87:18:34:d4:2b:60:3f:62:83:d9:0b:47:47:
         83:f9:f9:b0:ec:89:88:c2:0d:b3:d1:4c:f9:f5:0f:18:2a:27:
         13:a1:c4:56
```
Will result in this error.

The old cert looks like so:
```
 echo | openssl s_client -showcerts -servername pubsub.googleapis.com -connect pubsub.googleapis.com:443 2>/dev/null | openssl x509 -inform pem -noout -text
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            2b:49:5d:8b:ab:b8:3f:ec:1a:b3:24:64:d4:11:58:2f
    Signature Algorithm: sha256WithRSAEncryption
        Issuer: C=US, O=Google Trust Services, CN=Google Internet Authority G3
        Validity
            Not Before: Jul 29 18:38:19 2019 GMT
            Not After : Oct 21 18:23:00 2019 GMT
        Subject: C=US, ST=California, L=Mountain View, O=Google LLC, CN=edgecert.googleapis.com
        Subject Public Key Info:
            Public Key Algorithm: id-ecPublicKey
                Public-Key: (256 bit)
                pub:
                    04:01:73:85:97:7c:50:e8:de:ec:b6:46:ab:73:a5:
                    ad:b1:54:b0:a7:d2:17:4e:2b:46:48:d3:f0:1b:04:
                    03:ee:c8:c1:6e:66:88:c0:4d:1a:b3:66:d2:5f:a0:
                    ce:0e:ce:db:6a:86:45:03:e2:85:3a:84:ea:1d:f0:
                    2b:9d:25:6c:a5
                ASN1 OID: prime256v1
                NIST CURVE: P-256
        X509v3 extensions:
            X509v3 Extended Key Usage:
                TLS Web Server Authentication
            X509v3 Key Usage: critical
                Digital Signature
            X509v3 Subject Alternative Name:
                DNS:edgecert.googleapis.com, DNS:android.googleapis.com, DNS:bigtable.googleapis.com, DNS:clouderrorreporting.googleapis.com, DNS:cloudtrace.googleapis.com, DNS:container.googleapis.com, DNS:dataflow.googleapis.com, DNS:datastore.googleapis.com, DNS:fcm.googleapis.com, DNS:firestore.googleapis.com, DNS:logging.googleapis.com, DNS:monitoring.googleapis.com, DNS:pubsub.googleapis.com, DNS:runtimeconfig.googleapis.com, DNS:servicecontrol.googleapis.com, DNS:spanner.googleapis.com, DNS:stackdriver.googleapis.com
            Authority Information Access:
                CA Issuers - URI:http://pki.goog/gsr2/GTSGIAG3.crt
                OCSP - URI:http://ocsp.pki.goog/GTSGIAG3

            X509v3 Subject Key Identifier:
                41:83:85:54:71:86:1E:89:23:1A:E4:4B:58:8D:CA:C8:90:0A:0B:66
            X509v3 Basic Constraints: critical
                CA:FALSE
            X509v3 Authority Key Identifier:
                keyid:77:C2:B8:50:9A:67:76:76:B1:2D:C2:86:D0:83:A0:7E:A6:7E:BA:4B

            X509v3 Certificate Policies:
                Policy: 1.3.6.1.4.1.11129.2.5.3
                Policy: 2.23.140.1.2.2

            X509v3 CRL Distribution Points:

                Full Name:
                  URI:http://crl.pki.goog/GTSGIAG3.crl

    Signature Algorithm: sha256WithRSAEncryption
         28:61:c4:c8:97:67:7e:86:0e:78:b2:60:af:a2:e0:f6:d9:c3:
         ce:dd:d4:b6:06:df:24:f5:42:58:57:47:d9:11:b7:55:0c:a4:
         8d:c0:3a:07:95:f6:0e:c5:48:a7:2e:43:37:7b:b6:b5:3f:6c:
         17:21:03:d3:7a:c9:70:90:0f:4d:ac:11:5f:65:39:e1:99:e0:
         d8:0d:cf:28:3a:bb:4c:4f:fd:08:39:1e:24:59:f4:61:74:63:
         1e:72:8a:07:a3:80:c0:bc:14:d2:f7:4e:1c:e2:6f:00:58:fc:
         cf:88:39:da:60:6e:b3:14:24:15:40:55:0a:fe:5f:c6:ca:fa:
         ad:1f:36:cb:a9:5d:1c:5f:ee:aa:6e:dc:26:68:cc:c3:03:ae:
         15:cd:68:b5:2f:a0:dc:f2:02:b5:7f:c3:31:51:7e:ca:f2:b7:
         15:ca:77:ec:97:b3:49:03:84:ae:fe:cd:72:64:28:0b:d6:41:
         9b:f9:ee:c1:1f:d6:11:5c:e5:0d:f3:62:48:bf:84:9c:3a:b3:
         a2:55:62:96:3e:4d:9e:c6:aa:82:83:6d:6f:a7:6d:15:5d:31:
         c9:6c:88:89:14:78:07:55:b0:7e:90:c9:ef:02:cd:b3:e3:24:
         d0:9b:32:7e:57:e9:6f:be:1b:41:a5:32:bc:3f:cf:72:f8:51:
         44:b9:12:6b
```
And doesn't result in any error.

Ideally, it would also be easier to add more truststores too, so we could fall back to the host's ca certs, for example, but this is restricted at the moment by the way the config is loaded:
https://github.com/akka/alpakka/blob/b2d799be44551c8155e92f45b1173d2e6cd0184c/google-cloud-pub-sub-grpc/src/main/scala/akka/stream/alpakka/googlecloud/pubsub/grpc/impl/AkkaGrpcSettings.scala#L24

I'm happy to contribute to the configuration aspect of this, but I am not sure where the original `.crt` is from, nor how to update it.

**Comment by 2m** [comment]
Thank you for the detailed report.

The bundled crt has been taken from https://pki.goog/ (GTS GIAG3) but including specifically that crt has been causing issues in Google Cloud Platform's internal networks: #1871 

As you have noticed using the system specific certificate from `ca-certificates.crt` solves the issue, and that should be the default behaviour of the Alpakka Google Cloud Pub/Sub gRPC connector.

For the users, who do not have up-to-date certificates in `ca-certificates.crt` the extension point should be the `ssl-config` configuration directly, rather than Alpakka connector configuration.

So to fix this:
* bundled `GoogleInternetAuthorityG3.crt` should be removed
* `rootCa` Alpakka Google Cloud Pub/Sub gRPC setting should be removed
* by default default Java certificate store should be used
* an example of how to add a custom certificate by adding ssl-config configuration should be added.

WDYT?

**Comment by aperkins1310** [comment]
No worries. Yup makes sense to me, the main issue trying to configure a cert from the system was the rootCa config overriding what was set in the ssl config, so removing that instantly makes it more flexible. Your other points all seem sensible too 👍

**Comment by andrea-zanetti** [comment]
Hi, 

I'm having the same problem and i stumbled upon this issue. Could you please provide a complete workaround? I've not found any documentation on how to configure custom certificates with alpakka.

Thank you in advance.

**Comment by andrea-zanetti** [comment]
If anyone is interested i found a really simple, even if temporary, workaround: it is enough to add to your project resources a valid certificate (for Google Pub/Sub we used GS Root R2, downloaded from https://pki.goog/) and change the rootCa name to that of the valid certificate.

**Comment by l15k4** [comment]
Guys this issue is getting serious https://github.com/akka/alpakka/pull/1872#issuecomment-547823326 . Is anyone working on this?

**Comment by ennru** [comment]
I've created #2107 to finally remove this invalid certificate. Please check it out and let me know if we can do even better.

**Comment by ennru** [comment]
Fixed with #2107
